src/web/core/src/test/java/org/geoserver/web/wicket/EnvelopePanelBoundingBoxTest.java

```diff
@@ -0,0 +1,109 @@
+import org.geotools.geometry.jts.ReferencedEnvelope3D;
+import org.geotools.referencing.CRS;
+import org.junit.Test;
+import org.opengis.referencing.FactoryException;
+
+public class EnvelopePanelBoundingBoxTest extends EnvelopePanelTest {
+
+    private static final String FORM = "form";
+
+    @Test
+    public void testInvalidBoundingBoxValuesEqual() throws FactoryException {
+        final ReferencedEnvelope e =
+                new ReferencedEnvelope3D(180, 180, 90, 90, 20, 20, CRS.decode("EPSG:7415"));
+        tester.startPage(new FormTestPage(id -> providePanel(e, id)));
+
+        tester.assertComponent(FORM, Form.class);
+
+        FormTester ft = tester.newFormTester(FORM);
+
+        ft.submit();
+
+        assertEquals(3, tester.getFeedbackMessages(null).size());
+    }
+
+    @Test
+    public void testInvalidBoundingBoxValuesLower() throws FactoryException {
+        final ReferencedEnvelope e =
+                new ReferencedEnvelope3D(180, 180, 90, 90, 20, 20, CRS.decode("EPSG:7415"));
+        tester.startPage(new FormTestPage(id -> providePanel(e, id)));
+
+        tester.assertComponent(FORM, Form.class);
+
+        FormTester ft = tester.newFormTester(FORM);
+
+        ft.setValue("panel:minX", "181");
+        ft.setValue("panel:minY", "91");
+        ft.setValue("panel:minZ", "21");
+
+        ft.submit();
+
+        assertEquals(3, tester.getFeedbackMessages(null).size());
+    }
+
+    @Test
+    public void testValidBoundingBoxValues() throws FactoryException {
+        final ReferencedEnvelope e =
+                new ReferencedEnvelope3D(180, 181, 90, 91, 20, 21, CRS.decode("EPSG:7415"));
+        tester.startPage(new FormTestPage(id -> providePanel(e, id)));
+
+        tester.assertComponent(FORM, Form.class);
+
+        FormTester ft = tester.newFormTester(FORM);
+
+        ft.submit();
+
+        assertEquals(0, tester.getFeedbackMessages(null).size());
+    }
+
+    /**
+     * Tests if validation works correctly with comma decimal separator on English locale
+     *
+     * @throws FactoryException
+     */
+    @Test
+    public void testValidBoundingBoxWithEnglishLocaleAndComma() throws FactoryException {
+        final ReferencedEnvelope e =
+                new ReferencedEnvelope3D(-60, -59, -58, -57, -56, -55, CRS.decode("EPSG:7415"));
+        tester.startPage(new FormTestPage(id -> providePanel(e, id)));
+
+        tester.assertComponent(FORM, Form.class);
+
+        FormTester ft = tester.newFormTester(FORM);
+
+        Session.get().setLocale(Locale.ENGLISH);
+
+        /*
+        Settings values by UI since comma as decimal separator is added by fields
+        and values provided by panel after construction use dot
+        */
+        ft.setValue("panel:maxX", "-59,01");
+        ft.setValue("panel:maxY", "-57,01");
+        ft.setValue("panel:maxZ", "-55,01");
+
+        ft.submit();
+
+        assertEquals(0, tester.getFeedbackMessages(null).size());
+    }
```

Comment:
Tested with italian locale and proposed change above, this behavior mixing two locales was not needed anymore: suggest removing the test.

Reply:
Removed